test/suites/standard/logging.robot

```diff
@@ -0,0 +1,93 @@
+*** Settings ***
+Documentation       Tests related to different logging levels
+
+Resource            ../../resources/common.resource
+Resource            ../../resources/systemd.resource
+Resource            ../../resources/microshift-config.resource
+Resource            ../../resources/microshift-process.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           configuration    etcd    restart    slow
+
+
+*** Variables ***
+${LEVELNORMAL}      SEPARATOR=\n
+...                 ---
+...                 debugging:
+...                 \ \ LogLevel: NORMAL
+
+${LEVELDEBUG}       SEPARATOR=\n
+...                 ---
+...                 debugging:
+...                 \ \ LogLevel: debug
+
+${LEVELUNKNOWN}     SEPARATOR=\n
+...                 ---
+...                 debugging:
+...                 \ \ LogLevel: Normal
+
+
+*** Test Cases ***
+Test LogLevel Normal
+    [Documentation]    Set LogLevel to NORMAL
+    ...
+    ...    Test various spellings of the LogLevelkeyword
+    ...    uppercase, lowercase, Capitol, and camelcase are supported
+    [Setup]    Setup With Custom Config    ${LEVELNORMAL}
+    Expect LogLevel    NORMAL
+
+Test LogLevel Debug
+    [Documentation]    Set LogLevel to debug
+    ...
+    ...    Test various spellings of the logLevelkeyword
+    ...    uppercase, lowercase, Capitol, and camelcase are supported
+    [Setup]    Setup With Custom Config    ${LEVELDEBUG}
+    Expect LogLevel    debug
+
+Test LogLevel UNKNOWN
+    [Documentation]    Set logLevel to unknown
+    ...
+    ...    Test various spellings of the logLevelkeyword
+    ...    uppercase, lowercase, Capitol, and camelcase are supported
+    [Setup]    Setup With Custom Config    ${LEVELUNKNOWN}
+    Expect LogLevel    Normal
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig    # for readiness checks
+    Save Default MicroShift Config
+
+Teardown
+    [Documentation]    Test suite teardown
+    Restore Default Config
+    Logout MicroShift Host
+    Remove Kubeconfig
+
+Restore Default Config
+    [Documentation]    Remove any custom config and restart MicroShift
+    Restore Default MicroShift Config
+    Restart MicroShift
+
+Setup With Custom Config
+    [Documentation]    Install a custom config and restart MicroShift
+    [Arguments]    ${config_content}
+    ${merged}=    Extend MicroShift Config    ${config_content}
+    Upload MicroShift Config    ${merged}
+    Restart MicroShift
+
+Expect LogLevel
+    [Documentation]    Verify that the logLevel matches the expected value
+    [Arguments]    ${expected}
+
+    ${output}    ${stdout}=    Execute Command
+    ...    microshift show-config
```

Comment:
The show-config command reads the file and reports the contents, so it's not strictly necessary to restart microshift as part of these tests. If you wanted to ensure that the microshift service picked up the settings correctly after a restart, you could use some the keywords we have for watching the logs and picking out patterns. test/suites/standard/log-scan.robot has some examples and the keywords are defined in test/resources/journalctl.py